org.eclipse.xtext.ide.tests/src/org/eclipse/xtext/ide/tests/server/RenamePositionTest.xtend

```diff
@@ -0,0 +1,119 @@
+/*******************************************************************************
+ * Copyright (c) 2017 TypeFox GmbH (http://www.typefox.io) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.ide.tests.server
+
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.RenameParams
+import org.eclipse.lsp4j.TextDocumentIdentifier
+import org.eclipse.xtext.testing.AbstractLanguageServerTest
+import org.junit.Assert
+import org.junit.Test
+
+/**
+ * @author koehnlein - Initial contribution and API
+ */
+class RenamePositionTest extends AbstractLanguageServerTest {
+	
+	new() {
+		super('renametl')
+	}
+	
+	@Test
+	def void testRenameBeginningOfFile() {
+		val model = 'type Test'
+		val modelFile = 'MyType.testlang'.writeFile(model)
+        initialize
+        val position = new Position(0, 0)
+		val params = new RenameParams(new TextDocumentIdentifier(modelFile), position, 'Tescht')
+        val workspaceEdit = languageServer.rename(params).get
+        assertEquals('''
+			changes :
+				MyType.testlang : Tescht [[0, 5] .. [0, 9]]
+			documentChanges : 
+        '''.toString, toExpectation(workspaceEdit))	
+	}
+
+	@Test
+	def void testRenameEndOfFile() {
+		val model = 'type Test'
+		val modelFile = 'MyType.testlang'.writeFile(model)
+        initialize
+        val position = new Position(0, model.length)
+		val params = new RenameParams(new TextDocumentIdentifier(modelFile), position, 'Tescht')
+        val workspaceEdit = languageServer.rename(params).get
+        assertEquals('''
+			changes :
+				MyType.testlang : Tescht [[0, 5] .. [0, 9]]
+			documentChanges : 
+        '''.toString, toExpectation(workspaceEdit))	
+	}
+
+	@Test
+	def void testBeyondBeginningOfFile() {
+		val model = 'type Test'
+		val modelFile = 'MyType.testlang'.writeFile(model)
+        initialize
+        val position = new Position(-1, 0)
+		val params = new RenameParams(new TextDocumentIdentifier(modelFile), position, 'Tescht')
+        try {
+        	languageServer.rename(params).get
+        	Assert.fail('Rename should have failed')
```

Comment:
Instead of accepting an exception for bogus renames, shouldn't we support [`textDocument/prepareRename`](https://github.com/eclipse/xtext-core/pull/1051)?

This is language specific, but the [first test case's position](https://github.com/eclipse/xtext-core/pull/1084/files#diff-c5ae67f00096ce11d1e7e9c8396221c9R31) also looks odd to me; when the cursor position is on the `type` keyword, I would not expect a successful rename as a user.

Reply:
This has been changed.